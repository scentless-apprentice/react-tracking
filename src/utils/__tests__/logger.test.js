global.console.log = jest.fn();

const errors = [
  {
    keyword: 'additionalProperties',
    dataPath: '',
    message: 'should NOT have additional properties',
    params: {
      additionalProperty: 'someParam',
    },
  },
  {
    keyword: 'required',
    dataPath: '.thing',
    message: "should have required property 'id'",
    params: {
      additionalProperty: 'someParam',
    },
  },
  {
    keyword: 'type',
    dataPath: '.thing',
    message: 'should be number',
  },
];

describe('logger', () => {
  afterEach(() => {
    jest.resetModules();
    global.console.log.mockClear();
  });

  it('logs errors to console.log', () => {
    const logger = require('../logger'); // eslint-disable-line global-require
    logger.error(errors);

    expect(global.console.log).toHaveBeenCalled();
    expect(global.console.log.mock.calls).toMatchSnapshot();
  });

  it('uses console.group when available', () => {
    global.console.groupCollapsed = jest.fn();
    global.console.groupEnd = jest.fn();
    const logger = require('../logger'); // eslint-disable-line global-require

    logger.error(errors);

    expect(global.console.groupCollapsed).toHaveBeenCalledTimes(1);
    expect(global.console.groupEnd).toHaveBeenCalledTimes(1);
  });
});
