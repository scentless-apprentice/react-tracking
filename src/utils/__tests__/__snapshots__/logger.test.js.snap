// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger logs errors to console.log 1`] = `
Array [
  Array [
    "[tracking-schema] ðŸš« tracking data is invalid ðŸš«",
  ],
  Array [
    " ðŸ‘‰  [additionalProperties error]: ROOT should NOT have additional properties (it had 'someParam')",
  ],
  Array [
    " ðŸ‘‰  [required error]: ROOT.thing should have required property 'id'",
  ],
  Array [
    " ðŸ‘‰  [type error]: ROOT.thing should be number",
  ],
]
`;
